projects/plugins/jetpack/modules/subscriptions/subscribe-modal/subscribe-modal.php

```diff
@@ -0,0 +1,196 @@
+			),
+			'label'  => array(
+				'id'    => array(),
+				'for'   => array(),
+				'class' => array(),
+			),
+			'input'  => array(
+				'id'          => array(),
+				'type'        => array(),
+				'name'        => array(),
+				'value'       => array(),
+				'class'       => array(),
+				'style'       => array(),
+				'placeholder' => array(),
+				'required'    => array(),
+			),
+			'button' => array(
+				'type'  => array(),
+				'class' => array(),
+				'style' => array(),
+			),
+		);
+
+		?>
+			<div id="wpcom-subscribe-modal">
+				<div class="modal-content">
+					<?php
+					foreach ( $blocks as $block ) {
+						echo wp_kses( render_block( $block ), $allowed_html );
+					}
+					?>
+					<div id="close"><a href="#">Close</a></div>
+				</div>
+			</div>
+		<?php
+	}
+}
+add_action( 'wp_footer', 'wpcom_add_subscribe_modal_to_frontend' );
+
+/**
+ * Enqueues JS to load modal.
+ *
+ * @return void
+ */
+function wpcom_enqueue_subscribe_modal_assets() {
+	if ( should_enable_subscriber_modal() && ! is_admin() ) {
+		wp_enqueue_style( 'subscribe-modal-css', plugins_url( 'subscribe-modal.css', __FILE__ ), array(), JETPACK__VERSION );
+		wp_enqueue_script( 'subscribe-modal-js', plugins_url( 'subscribe-modal.js', __FILE__ ), array(), JETPACK__VERSION, true );
+	}
+}
+add_action( 'wp_enqueue_scripts', 'wpcom_enqueue_subscribe_modal_assets' );
+
+/**
+ * Returns true if we should enable Newsletter content.
+ * This is currently limited to lettre theme or newsletter sites.
+ * We could open it to all themes or site intents.
+ *
+ * @return bool
+ */
+function should_enable_subscriber_modal() {
+	$is_lettre          = get_option( 'stylesheet' ) === 'lettre';
+	$is_newsletter_site = get_option( 'site_intent' ) === 'newsletter';
+	$is_modal_enabled   = get_option( 'sm_enabled' ) || false;
+
+	return ( $is_lettre || $is_newsletter_site ) && $is_modal_enabled;
+}
+
+/**
+ * Returns the slug for the Subcribe template.
+ *
+ * @return string
+ */
+function get_subscribe_template_slug() {
+	return 'subscribe-modal';
+}
+
+/**
+ * Returns the title for the Subcribe template.
+ *
+ * @return string
+ */
+function get_subscribe_template_title() {
+	return 'Subscribe Modal';
+}
+
+/**
+ * Returns the initial content of the Subscribe Modal template.
+ * This can then be edited by the user.
+ *
+ * @return string
+ */
+function get_subscribe_template_content() {
+	return '<!-- wp:group {"style":{"spacing":{"padding":{"top":"var:preset|spacing|40","right":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|40"}}},"layout":{"type":"constrained"}} -->
+	<div class="wp-block-group" style="padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--40);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--40)"><!-- wp:heading {"textAlign":"center"} -->
+	<h2 class="wp-block-heading has-text-align-center">This post is for subscribers</h2>
```

Comment:
Translation function.

Reply:
Is there any weirdness about trying to use a translation function inside of block markup?

Reply:
Note: I've actually moved the text/markup for the Close button inside the template part now too, so same question will apply on that.

Reply:
Updated see https://github.com/Automattic/jetpack/pull/31393#discussion_r1244160182